listeners/test-client.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+var Mocha = require('mocha');
+var fs = require('fs');
+var path = require('path');
+var Q = require('q');
+
+class TestClient {
+
+    constructor () {
+      var self = this;
+      self._testResult = Q.defer();
+    }
+
+    _getTestResult(){
+      var self = this;
+      return self._testResult.promise;
+    }
+
+    _runTest(asrResult, speakerIds){
+      var self = this;
+
+      //  Get the variables set by main.rule
+      var params = asrResult.NLParse;
+      if (params.action === 'testing') {
+          var dir = '';
+          switch (params.type) {
+              case 'unit':
+                  dir = './test/unit/';
+                  break;
+
+              case 'func':
+                  dir = './test/func/';
+                  break;
```

Comment:
Let's add a default case here for error checking because right now Jibo will break if someone said nonsense like "run apple tests"

Behavior should be for Jibo to reply something like "unknown test" and resume idling